### DISCUSSION TITLE
Highlighting all sublayers when one is hovered over/picked?

### DISCUSSION THREAD

**Discussion by mcost45** [visgl/deck.gl#5882] [Q&A]
Hi, I have a composite layer formed from a path, solid polygon, and text layer, in order to create a 'text label with an arrow'. Currently the sublayers get picked and highlighted individually, but it would be preferable for each of the sublayers to be highlighted together, when one is hovered over. I understand this means each of the sublayers have separate picking colours, but is it possible to make them all share the same picking colour? It seems there is no way to work with custom picking colours from within the composite layer, but that it can only be done by extending the individual sublayers. Are there any suggested ways for handling this?

**Comment by Pessimistress** [comment]
In the 8.5.0-alpha release a composite layer automatically highlights all sub layers with the same picked index.

If you need to highlight different indices for each sub layer, the composite layer can trigger an update with `this.setState` in the `getPickingInfo` method and in the next render pass a custom `highlightedObjectIndex` prop to each sub layer.
